nucleus/security/ssl-impl/src/main/java/com/sun/enterprise/security/ssl/impl/SecuritySupportImpl.java

```diff
@@ -145,86 +157,327 @@ protected SecuritySupportImpl(boolean init) {
             initJKS();
         }
     }
+   
 
-    private void initJKS() {
-        String keyStoreFileName = null;
-        String trustStoreFileName = null;
+    // --- implements SecuritySupport ---
+    
+    /**
+     * This method returns an array of keystores containing keys and certificates.
+     */
+    @Override
+    public KeyStore[] getKeyStores() {
+        List<KeyStore> keyStoresList = keyStores.get(DEFAULT_MAP_KEY);
+        
+        return keyStoresList.toArray(new KeyStore[keyStoresList.size()]);
+    }
+    
+    /**
+     * This method returns an array of truststores containing certificates.
+     */
+    @Override
+    public KeyStore[] getTrustStores() {
+        List<KeyStore> trustStoresList = trustStores.get(DEFAULT_MAP_KEY);
+        
+        return trustStoresList.toArray(new KeyStore[trustStoresList.size()]);
+    }
+    
+    /**
+     * This method returns an array of token names in order corresponding to array of keystores.
+     */
+    @Override
+    public String[] getTokenNames() {
+        List<String> tokenNamesList = tokenNames.get(DEFAULT_MAP_KEY);
+        
+        return tokenNamesList.toArray(new String[tokenNamesList.size()]);
+    }
+    
+    /**
+     * @param token
+     * @return a keystore
+     */
+    @Override
+    public KeyStore getKeyStore(String token) {
+        int tokenIndex = getTokenIndex(token);
+        if (tokenIndex < 0) {
+            return null;
+        }
+
+        return keyStores.get(DEFAULT_MAP_KEY).get(tokenIndex);
+    }
+
+    /**
+     * @param token
+     * @return a truststore
+     */
+    @Override
+    public KeyStore getTrustStore(String token) {
+        int tokenIndex = getTokenIndex(token);
+        if (tokenIndex < 0) {
+            return null;
+        }
+
+        return trustStores.get(DEFAULT_MAP_KEY).get(tokenIndex);
+    }
+    
+    @Override
+    public void reset() {
+        
+        // Get store file names
+        String keyStoreFileName = System.getProperty(keyStoreProp);
+        String trustStoreFileName = System.getProperty(trustStoreProp);
+        
+        // Get store passwords
+        char[] keyStorePass = masterPasswordHelper.getMasterPassword();
+        char[] trustStorePass = keyStorePass;
+        
+        if (keyStorePass == null || isACC() || (serverEnvironment != null && serverEnvironment.isMicro())) {
+            String keyStorePassOverride = System.getProperty(KEYSTORE_PASS_PROP, DEFAULT_KEYSTORE_PASS);
+            if (keyStorePassOverride != null) {
+                keyStorePass = keyStorePassOverride.toCharArray();
+            }
+            
+            String trustStorePassOverride = System.getProperty(TRUSTSTORE_PASS_PROP, DEFAULT_TRUSTSTORE_PASS);
+            if (trustStorePassOverride != null) {
+                trustStorePass = trustStorePassOverride.toCharArray();
+            }
+        }
+        
+        // re-load stores
+        initStores(
```

Comment:
@MattGill98  
I was looking at that indeed, and initially I had it shared. But the initJKS method also does the instantiation of the dependencies if they aren't injected, and has the booleans that check for being initialised.

Since the methods work on multiple local variables and they have the boolean guard, it's not so nice to try to reuse it directly.

I'll nevertheless will try to give it a second shot.